components/dashboard/src/admin/TeamDetail.tsx

```diff
@@ -0,0 +1,105 @@
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import moment from "moment";
+import { useEffect, useState } from "react";
+import { Team, TeamMemberInfo, TeamMemberRole } from "@gitpod/gitpod-protocol";
+import { getGitpodService } from "../service/service";
+import { Item, ItemField, ItemsList } from "../components/ItemsList";
+import PillLabel from "../components/PillLabel";
+import DropDown from "../components/DropDown";
+import { Link } from "react-router-dom";
+
+export default function TeamDetail(props: { team: Team }) {
+    const { team } = props;
+    const [teamMembers, setTeamMembers] = useState<TeamMemberInfo[] | undefined>(undefined);
+    const [searchText, setSearchText] = useState<string>('');
+
+    useEffect(() => {
+        (async () => {
+            const members = await getGitpodService().server.adminGetTeamMembers(team.id);
+            if (members.length > 0) {
+                setTeamMembers(members)
+            }
+        })();
+    }, [team]);
+
+    const filteredMembers = teamMembers?.filter(m => {
+        const memberSearchText = `${m.fullName || ''}${m.primaryEmail || ''}`.toLocaleLowerCase();
+        if (!memberSearchText.includes(searchText.toLocaleLowerCase())) {
+            return false;
+        }
+        return true;
+    });
+
+    const setTeamMemberRole = async (userId: string, role: TeamMemberRole) => {
+        await getGitpodService().server.adminSetTeamMemberRole(team!.id, userId, role);
+        setTeamMembers(await getGitpodService().server.adminGetTeamMembers(team!.id));
+    }
+    return <>
+        <div className="flex">
+            <div className="flex-1">
+                <div className="flex"><h3>{team.name}</h3>
+                    {team.markedDeleted && <span className="mt-2"><PillLabel type="warn" className="font-semibold mt-2 py-0.5 px-2 self-center">Deleted</PillLabel></span>}
+                </div>
+                <p className="mb-6">/t/{team.slug}</p>
+                {!team.markedDeleted && <>
+                    <span className="text-gray-400">{teamMembers && teamMembers.length} member(s)</span>
+                    <span className="text-gray-400"> · </span>
+                </>}
+                <span className="text-gray-400">Created on {moment(team.creationTime).format('MMM D, YYYY')}</span>
+            </div>
+        </div>
+        <div className="flex mt-4">
+            <div className="flex">
+                <div className="py-4">
+                    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16" width="16" height="16"><path fill="#A8A29E" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" /></svg>
+                </div>
+                <input type="search" placeholder="Search Members" onChange={e => setSearchText(e.target.value)} />
```

Comment:
**suggestion**: Could we include an empty state here for the deleted teams that include no members as we did with prebuilds in https://github.com/gitpod-io/gitpod/pull/7882#discussion_r795668199?

Reply:
Yes - I tried this as well yesterday. What text should we show? "Members in deleted teams have been removed"?

Reply:
:a: Going with a boring (=simple) solution here like the following sounds ok. What do you think?

> No members found.

:b: Alternatively, we could not remove any members and keep the last known state of members.

Reply:
I'll go with A, since that's also the text we show when we can't find any members. 
B is unfortunately not possible, because unlike the "marked-deletion" of a team, when a team is marked as deleted, we actually delete all the team members. I vaguely remember a discussion being made about team members being hard-deleted, but I can't remember what or why that was 🤔